### DISCUSSION TITLE
Accessibility

### DISCUSSION THREAD

**Discussion by Trinitou** [free-audio/clap#225] [Ideas]
I'm no expert concerning accessibility. But it should indeed be possible to provide some accessibility features for plugins. Some basic ideas here:

- for **color blind** people: the host could request for certain color schemes. Is there a standard set of color schemes for color blind people?
- for **blind people**:
  - the plugin could handle screen reader output itself. But then there should be a possiblity to reserve an audio output for the screen reader. The host should then route this output only into monitoring but not into the audio master.
  - the plugin could provide access to the host for controlling the current focus GUI element:
    - some accessibility string for the focus element that the host can hand over to a screen reader
    - commands like jump to next/previous focus element, reset focus element
    - maybe the host could tell the plugin which keys
  - don't know if any blind people work with mouse but if so the plugin could notify the host about the current mouse hover GUI element

**Comment by abique** [comment]
Can't the plugin figure the accessibility requirement from the operating system?

**Comment by baconpaul** [comment]
So I wouldn't say I'm an expert, but Surge is probably the most accessible synth in the world and we have a big community of blind users. Along with reuk on the JUCE team, we were able to make (almost) the entire thing work with mac and windows screen readers.

Reaper and Reaper extensions work well, but some other DAWs have screen reader support. 

And I spent a lot of 2022 interacting with screen readers, accessibility and so on. At this point almost every surge feature is accessible, an that's true in the CLAP today! Run the CLAP in reaper with osara or in the standalone and voila screen reader all works. it's really cool actually!

We have a swath of blind users who make amazing music with surge and reaper, or the surge standalone, and who also use it in performance contexts, and the accessible team has contributed to the project in many ways.

But: This is nothing to do with CLAP. The point at which accessibility happens is well *above* the CLAP layer. The accessible APIs are OS specific, require detailed work which has nothing to do with the plug, and all happen after the HWND. Plus if you implement them your OS will detect them no matter the plugin interface. That's why they work in the CLAP today.

So I think this is a great idea, real work, totally feasible, but not part of the CLAP spec.
